src/Databases/DatabaseReplicated.cpp

```diff
@@ -585,7 +585,8 @@ void DatabaseReplicated::checkQueryValid(const ASTPtr & query, ContextPtr query_
 
         if (auto * create = query->as<ASTCreateQuery>())
         {
-            bool replicated_table = create->storage && create->storage->engine && startsWith(create->storage->engine->name, "Replicated");
+            bool replicated_table = create->storage && create->storage->engine &&
```

Comment:
Why don't add a method `isReplicated` to `IStorage`?

Reply:
There's no instance of `IStorage` here. But maybe we can add it to `StorageFactory`